example/extended_ml_workflow.py

```diff
@@ -0,0 +1,130 @@
+"""
+An extended workflow of the toy ML workflow as defined in `example.ml_workflow`.
+The extended example can also return the ROC AUC score and allows users to switch
+to a larger UCI ML covertype dataset, which can be found at:
+http://archive.ics.uci.edu/ml/datasets/Covertype
+"""
+
+import pandas as pd
+from sklearn import datasets, metrics
+from example.ml_workflow import flow as basic_flow
+import time
+
+import bionic as bn
+
+# Convert the example ml workflow to a builder we'll use for extending.
+builder = basic_flow.to_builder()
+
+
+# Predict probabilities for the test data.
+@builder
+def prediction_frame(model, test_frame):
+    """
+    A dataframe with probability predictions for the test data.
+    """
+    predictions = model.predict_proba(test_frame.drop("target", axis=1))[:, 1]
+    df = pd.DataFrame()
+    df["proba"] = predictions
+    return df
+
+
+# Evaluate the model's precision and recall over a range of threshold values.
+@builder
+def precision_recall_frame(test_frame, prediction_frame):
+    """
+    A dataframe with three columns:
+    - `threshold`: a probability threshold for the model
+    - `precision`: the test set precision resulting from that threshold
+    - `recall`: the test set recall resulting from that threshold
+    """
+    precisions, recalls, thresholds = metrics.precision_recall_curve(
+        test_frame["target"], prediction_frame["proba"],
+    )
+
+    df = pd.DataFrame()
+    df["threshold"] = [0] + list(thresholds) + [1]
+    df["precision"] = list(precisions) + [1]
+    df["recall"] = list(recalls) + [0]
+
+    return df
+
+
+# Summarize the model's performance in one number.
+@builder
+def auc_score(test_frame, prediction_frame):
```

Comment:
This capitalization is confusing, since it doesn't add up to "AUC". Maybe put parentheses around "Receiver Operating Characteristic"?